src/reconcile/rec_visibility.c

```diff
@@ -245,11 +249,20 @@ __rec_find_and_save_delete_hs_upd(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_
                 break;
             }
         }
+
+        /* Track the first globally visible self-contained value. */
+        if (!F_ISSET(r, WT_REC_EVICT) && visible_all_upd == NULL &&
+          __wt_txn_upd_visible_all(session, delete_upd) && WT_UPDATE_DATA_VALUE(delete_upd))
+            visible_all_upd = delete_upd;
     }
 
     WT_ASSERT_ALWAYS(session, delete_tombstone == NULL || delete_upd != NULL,
       "If we delete a tombstone from the history store, we must also delete the update.");
 
+    /* Free obsolete updates if exist. */
+    if (FLD_ISSET(S2C(session)->strategy_flags, WT_CONN_STRATEGY_OBSOLETE_CHECK) &&
+      !delete_hs_upd_found && visible_all_upd != NULL && visible_all_upd->next != NULL)
+        __wt_free_update_next_list(session, visible_all_upd);
```

Comment:
This doesn't return the old behavior. The old behavior was based on the global timestamp moving between time comparisons in a running system. I don't think it was meaningfully linked to reconciliation.